### PR TITLE
Support ed25519 signature 2020

### PR DESCRIPTION
**Pull request by JSAssassin**

### CODE REVIEW

File: tests/10-EdvClient.spec.js [hunk 1]
```diff
@@ -6,9 +6,10 @@ import mock from './mock.js';
 import {isRecipient} from './test-utils.js';
 
 describe('EdvClient', () => {
-  let invocationSigner, keyResolver = null;
+  let invocationSigner, keyResolver, kAK = null;
```
Comment (mattcollier) [inline tests/10-EdvClient.spec.js:9]: I see this is a carry over, but style prohibits initializing multiple variables on the same line.

File: tests/10-EdvClient.spec.js [hunk 2]
```diff
@@ -6,9 +6,10 @@ import mock from './mock.js';
...
   before(async () => {
     await mock.init();
+    kAK = mock.keys.keyAgreementKey;
```
Comment (mattcollier) [inline tests/10-EdvClient.spec.js:12]: this is a similar situation to the acronym URL.  In these cases I think it's best to just make it all lower case.  So in this case `kak`.

File: tests/15-EdvClientRevokeCapability.spec.js [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
+import {createRecipient, JWE_ALG} from './test-utils.js';
 import {EdvClient} from '..';
+import {Ed25519Signature2020} from '@digitalbazaar/ed25519-signature-2020';
+import jsigs from 'jsonld-signatures';
 import mock from './mock.js';
-const {SECURITY_CONTEXT_V2_URL, sign, suites: {Ed25519Signature2018}} = jsigs;
-import {CapabilityDelegation} from 'ocapld';
-import {createRecipient, JWE_ALG} from './test-utils.js';
+import uuid from 'uuid-random';
+import {constants} from 'ed25519-signature-2020-context';
```
Comment (mattcollier) [inline tests/15-EdvClientRevokeCapability.spec.js:11]: This is correct. However, the latest `Ed25519Signature2020`does export the context as well.  If this export is used, this can be simplified and the `ed25519-signature-2020-context` dependency can likely be removed.  It would be good to update the pattern used here for demonstration purposes.

https://github.com/digitalbazaar/ed25519-signature-2020/blob/main/lib/Ed25519Signature2020.js#L198-L199

File: tests/test-utils.js [hunk 4]
```diff
@@ -39,7 +33,13 @@ export function isNewEDV({inserted, testId, hmac, sequence = 0}) {
...
 // recipient should be JOSE
 // @see https://tools.ietf.org/html/rfc8037
-export function isRecipient({recipient, expected = _x25519Expected}) {
+export function isRecipient({recipient}) {
+  const expected = {
+    kid: mock.keys.keyAgreementKey.id,
+    alg: JWE_ALG,
+    crv: 'X25519',
+    kty: 'OKP'
+  };
```
Comment (mattcollier) [inline tests/test-utils.js:42]: +1 to the clean-up you did here.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,11 @@
 # edv-client ChangeLog
 
+## 8.1.0 - 2021-04-TBD
```
Comment (mattcollier) [inline CHANGELOG.md:3]: Any breaking change means a major release.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,11 @@
...
+
+### Changed
+- **BREAKING**: Use (@digitalbazaar/minimal-cipher@3.0)[https://github.com/digitalbazaar/minimal-cipher/blob/master/CHANGELOG.md]
```
Comment (mattcollier) [inline CHANGELOG.md:6]: Please add subpoint: Now uses `X25519KeyAgreementKey2020` instead of `X25519KeyAgreementKey2020`.

**Review by mattcollier — CHANGES_REQUESTED** [review]

File: tests/10-EdvClient.spec.js (hunk 1 above)
Comment (JSAssassin) [reply tests/10-EdvClient.spec.js]: Fixed 437f761

File: tests/10-EdvClient.spec.js (hunk 2 above)
Comment (JSAssassin) [reply tests/10-EdvClient.spec.js]: Updated 437f761

File: tests/15-EdvClientRevokeCapability.spec.js (hunk 3 above)
Comment (JSAssassin) [reply tests/15-EdvClientRevokeCapability.spec.js]: Updated 082e3e9  & e57ddd9

File: CHANGELOG.md (hunk 6 above)
Comment (JSAssassin) [reply CHANGELOG.md]: Updated.

File: package.json [hunk 7]
```diff
@@ -65,11 +67,12 @@
     "mocha": "^8.3.2",
     "mocha-lcov-reporter": "^1.3.0",
     "node-forge": "^0.10.0",
+    "npm": "^7.8.0",
```
Comment (dlongley) [inline package.json:70]: This looks suspicious to me. Can this be removed, why is this here?

**Review by dlongley — COMMENTED** [review]
Will this still support docs that were previously encrypted using kak 2019? If the default key associated with an EDV is a 2019 key can this client continue to use it? If not, why not -- and how much of a lift would it be to support that?

File: package.json (hunk 7 above)
Comment (JSAssassin) [reply package.json]: Thanks for catching this. It was a typo, I've removed it.

**Comment by dmitrizagidulin** [comment]
@dlongley @JSAssassin - Yes, documents previously encrypted using kak 2019 are compatible with kak 2020 (they can be decrypted etc, no problem). Here's an example backwards-compatibility unit test that demonstrates this:

https://github.com/digitalbazaar/x25519-key-agreement-key-2020/blob/main/test/X25519KeyAgreementKey2020.spec.js#L159

**Comment by dlongley** [comment]
@dmitrizagidulin -- Just to get further clarification, can you say whether or not this scenario will work:

1. Create an EDV w/config that has a 2019 kak using the old edv-client.
2. Write some docs to the EDV using the 2019 kak and the old edv-client.
3. Read some docs from the EDV using the 2019 kak and the new edv-client.
4. Write some docs to the EDV using the 2019 kak and the new edv-client such that the old edv-client can read them.

Will a new edv-client connecting to an old EDV with this sort of configuration just continue to work and not create incompatibilities for old edv-clients?

**Review by dmitrizagidulin — APPROVED** [review]
:+1: 
Thanks @JSAssassin, good work here.

**Review by dlongley — APPROVED** [review]
Approving based on decisions made around not needing backwards compatibility.

**Comment by mattcollier** [comment]
@JSAssassin please let me know when this has been tested throughout the stack, specifically in `bedrock-kms-http`.

**Review by mattcollier — APPROVED** [review]
TY.
